horizon: 100 # 单位：步
dt: 0.05 # 单位：秒
substep: 0.01 # 单位：秒
x_weight : [100, 100, 10, 1, 0.1, 1] # 状态变量权重
lambda : 0.5 # 正则化/学习率参数
alpha : 0.1 # 探索参数
iter: 4 # 滚动优化迭代次数
heartbeat_duration: 0.1 # 控制器心跳周期
stddev: 10.0 # 控制器输出噪声标准差
input_limit: 20.0 # 控制器输入限制

dynamics_test: # 以实际动力学为准
  left_input: 0 # 左推力输入
  right_input: 10 # 右推力输入
hydrodynamics: # 水动力学参数
  mass: 38.0
  Iz: 6.25
  B: 0.74
  Xu_dot: -1.9
  Yv_dot: -29.3171
  Nr_dot: -4.2155
  Xu: 26.43
  Yv: 72.64
  Nr: 22.96

fxtdo:
  L1: [0.3, 2.2, 2.5]       # u 方向减小收敛增益
  L2: [0.7, 3.5, 3.5]       # u 方向减小扰动更新增益
  k1: [0.5, 2.2, 2.2]       # u 方向减小非线性响应
  k1p: [0.05, 0.35, 0.35]    # u 方向减小线性增益
  k1pp: [0.2, 1.1, 1.1]     # u 方向减小高阶非线性
  k2: [4.5, 4.5, 4.5]
  k2p: [0.8, 0.8, 0.8]
  k2pp: [0.7, 0.7, 0.7]
  d_inf: [0.05, 0.3, 0.3]

topics:
  observation: /heron/odom
  target: /heron/goal
  left_thruster_cmd: /heron/left_thruster_cmd
  right_thruster_cmd: /heron/right_thruster_cmd
simulation:
  speed_scale: 2.0 # 速度缩放因子
  time_total: 250 # 仿真总时长
  # 机动圆
  radius: 10 # 圆轨迹半径
  omega: 0.06 # 圆轨迹角速度
  amplitude_x: 12
  amplitude_y: 20
  angular_x: 0.05
  angular_y: 0.025
  x_goal: 50 # 点轨迹目标x坐标
  y_goal: 50 # 点轨迹目标y坐标
  psi_goal: 3.14159 # 点轨迹目标航向
